contracts/package.json

```diff
@@ -28,7 +28,7 @@
   },
   "license": "BSD-3-Clause",
   "scripts": {
-    "build": "rm -rf dist/ && tsc -b",
+    "build": "rm -rf dist/ && tsc -b && npx hardhat compile --network local",
```

Comment:
curious why this is needed?

Reply:
I was debugging an issue where the hardhat compile would just give up with no error, so I moved the compilation earlier. I thought it was because we would using the compiler in parallel.

This was the issue though https://github.com/NomicFoundation/hardhat/issues/3877